src/components/Countries/SubComponent/SearchFilter/SearchFilter.tsx

```diff
@@ -0,0 +1,33 @@
+'use client'
+import { useRouter } from 'next/navigation'
+import { ChangeEvent, FormEvent, useState } from 'react'
+import Style from './SearchFilter.module.scss'
+import SearchSvgIcon from '@/components/common/svg/SearchSvgIcon'
+
+export default function SearchFilter() {
+  const router = useRouter()
+  const [search, setSearch] = useState('')
+  const handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSearch(e.target.value)
+  }
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    router.push(`?search=${search}`)
```

Comment:
I think it might be better if the URL was like this. What do you think?

```search?query=${search}```

Reply:
I agree it would make it better. But this would be another page in which will contain the same codes. This will be more suitable if there is UI for search page.